### PR TITLE
fix: Retry RPC to the core node

### PR DESCRIPTION
**Pull request by ieQu1**

### CODE REVIEW

File: src/mria_config.erl [hunk 1]
```diff
@@ -109,6 +111,14 @@ whoami() ->
 rpc_module() ->
     persistent_term:get(?mria(rlog_rpc_module), gen_rpc).
 
+-spec core_rpc_retries() -> integer().
+core_rpc_retries() ->
+    persistent_term:get(?mria(core_rpc_retries), 10).
+
+-spec core_rpc_cooldown() -> integer().
+core_rpc_cooldown() ->
+    persistent_term:get(?mria(core_rpc_cooldown), 10).
```
Comment (thalesmg) [inline src/mria_config.erl:120]: Maybe add a comment or doc specifying in which unit is this?

Where it is [used](https://github.com/emqx/mria/pull/89/files#diff-f715566e94b14bb23ba14b99cee05368a7b2011b27ae6be37db988fbe56482efR289), looks like it'll sleep for only 10 ms, but `mria_lb`'s probe period is 1 s.

File: src/mria_lb.erl [hunk 2]
```diff
@@ -217,7 +223,8 @@ ping_core_nodes(NewCoreNodes) ->
     LocalMember = mria_membership:make_new_local_member(),
     lists:foreach(
       fun(Core) ->
-              mria_membership:ping(Core, LocalMember)
+              mria_membership:ping(Core, LocalMember),
+              mria_membership:monitor(membership, self(), true)
```
Comment (thalesmg) [inline src/mria_lb.erl:227]: Would it be safer to start the monitor before pinging to avoid losing the response?

Comment (ieQu1) [reply src/mria_lb.erl:227]: This is a foreach, so the responce doesn't matter. We ping first to make sure the shard server is already up and running.

File: src/mria_config.erl (hunk 1 above)
Comment (ieQu1) [reply src/mria_config.erl:120]: Fixed.

File: src/mria_lb.erl (hunk 2 above)
Comment (thalesmg) [reply src/mria_lb.erl:227]: I meant not the response from the `mria_membership:monitor` call, but the message/info that `mria_membership/mria_node_monitor` will send as a message later.

There could be a race condition where we ping, the other side emits the response, but the `mria_lb` process didn't get yet to monitor that, maybe?

**Review by thalesmg — APPROVED** [review]
Q: is the intention indeed to sleep only 10 ms between rpc retries while the `mria_lb` period is 1 s?  According to [this comment](https://github.com/emqx/mria/pull/89/files#diff-f715566e94b14bb23ba14b99cee05368a7b2011b27ae6be37db988fbe56482efR288), it may have not enough time to do more probing?  :thinking:

File: src/mria_lb.erl (hunk 2 above)
Comment (ieQu1) [reply src/mria_lb.erl:227]: Not sure that I understand you... `mria_membership:monitor` does a synchronous call to the local `gen_server` that adds an entry (what kind of events should be sent and where) to its state. So we start monitoring the node regardless of its state. 
mria_lb only reacts on the down events.

**Comment by ieQu1** [comment]
> Q: is the intention indeed to sleep only 10 ms between rpc retries while the `mria_lb` period is 1 s? According to [this comment](https://github.com/emqx/mria/pull/89/files#diff-f715566e94b14bb23ba14b99cee05368a7b2011b27ae6be37db988fbe56482efR288), it may have not enough time to do more probing? thinking

My initial intention was to rely on the node down event that the LB should receive. But this is a fair comment, I will try to make it more reliable

File: src/mria_lb.erl (hunk 2 above)
Comment (thalesmg) [reply src/mria_lb.erl:227]: You are right, I got confused and thought for a moment that the node event would depend on the monitor being set first.  Ignore me.

**Review by thalesmg — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

**Pull request merged**